main.c

```diff
@@ -111,9 +116,17 @@ static void __interrupt() interrupt_handler() {
 // This is called from within can_handle_interrupt()
 static void can_msg_handler(const can_msg_t *msg) {
     uint16_t msg_type = get_message_type(msg);
+    int cmd_type = -1;
     switch (msg_type) {
         case MSG_GENERAL_CMD:
-            // nothing right now
+            // toggle ON_OFF to turn off module properly
+            cmd_type = get_general_cmd_type(msg);
+            if (cmd_type == BUS_DOWN_WARNING) {
+                LATC2 = 1;
```

Comment:
When I said that if the shutdown sequence involved any delays it shouldn't go in the ISR, this is what I was referring to. While you perform those delays, you can't receive any other interrupt, so you can miss things. Since this is the shutdown sequence and you don't really care about anything that happens after this, I don't think it's worth fixing given the time constraints. For next time though, this is something to keep in mind and avoid doing.